### PR TITLE
Move query adapter from canonical-reports

### PR DESCRIPTION
**Pull request by ypc-faros**
## Description

Move query adapter and syncs graphql functionality with the latest changes from canonical-reports

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: test/tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,9 @@
   "compilerOptions": {
     "outDir": "../out/test",
   },
+  "files": [
+    "../node_modules/jest-expect-message/types/index.d.ts"
```
Comment (tovbinm) [inline test/tsconfig.json:7]: what is this for?

File: src/adapter/types.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    if (gql.isLeafType(type)) {
+      return {entryType: asPrimitiveType(type)};
+    }
+  } else if (gql.isLeafType(type)) {
+    return asPrimitiveType(type);
+  }
+  throw new VError('unknown GraphQL leaf type: %s', type);
+}
+
+// TODO: Merge with graphql.ts. This is implemented differently.
```
Comment (tovbinm) [inline src/adapter/types.ts:103]: does the TODO still apply?

**Review by tovbinm — APPROVED** [review]

File: test/tsconfig.json (hunk 1 above)
Comment (eskrm) [reply test/tsconfig.json:7]: It's part of the Typescript setup for this library: https://github.com/mattphillips/jest-expect-message#configure-typescript

**Review by eskrm — APPROVED** [review]
LGTM.

One nitpick: would be nice to put the schemas in `test/resources` under their own directory.

File: src/adapter/types.ts (hunk 2 above)
Comment (ypc-faros) [reply src/adapter/types.ts:103]: Yes, it still applies

Comment (ypc-faros) [reply src/adapter/types.ts:103]: I'll look into it and address in another PR

**Pull request merged**